lib/Runtime/Language/TaggedInt.cpp

```diff
@@ -452,17 +502,18 @@ namespace Js
 
     JavascriptString* TaggedInt::ToString(int value, ScriptContext* scriptContext)
     {
-        char16 szBuffer[20];
-        ToBuffer(value, szBuffer, _countof(szBuffer));
+        char16 szBuffer[22];
+        int pos = ToBuffer(value, szBuffer, _countof(szBuffer));
 
-        return JavascriptString::NewCopySz(szBuffer, scriptContext);
+        return JavascriptString::NewCopyBuffer(szBuffer + pos, (_countof(szBuffer) - 1) - pos, scriptContext);
     }
+
     JavascriptString* TaggedInt::ToString(uint value, ScriptContext* scriptContext)
     {
-        char16 szBuffer[20];
-        ToBuffer(value, szBuffer, _countof(szBuffer));
+        char16 szBuffer[22];
```

Comment:
I prefer the force-check under new toString implementation is hard coded instead of define once for all. Since that one is hard coded and doesn't allow anything below for known architectures, remaining changes wouldn't go beyond the make-up? 

p.s.: I don't have a strong opinion on make-up part but would like to keep buffer checking hard coded. There is enough explanation there.